### PR TITLE
Fix delay in KYA and Quiz translation

### PR DESCRIPTION
**Pull request by BenjaminSsempala**
#### Summary of Changes (What does this PR do?)

- [x] Remove lag in translation of KYA and Quiz
- [x] Fixes for KYA progress bar

#### Status of maturity (all need to be checked before merging):

- [ ] I've tested this locally
- [ ] I consider this code done
- [ ] This change ready to hit production in its current state
- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?

- Closes #<enter issue number here>
- [Jira_card_number]() (If exists)

### CODE REVIEW

File: mobile/lib/services/firebase_service.dart [hunk 1]
```diff
@@ -14,10 +14,18 @@ import 'package:flutter/cupertino.dart';
 
 class CloudAnalytics {
   static Future<bool> logEvent(CloudAnalyticsEvent analyticsEvent) async {
-    await FirebaseAnalytics.instance.logEvent(
-      name: analyticsEvent.snakeCase(),
-    );
+    try {
```
Comment (BenjaminSsempala) [inline mobile/lib/services/firebase_service.dart:17]: Just adding error Handling here

File: mobile/lib/screens/dashboard/dashboard_view.dart [hunk 2]
```diff
@@ -573,9 +571,6 @@ class _DashboardViewState extends State<DashboardView>
 
     context.read<FavouritePlaceBloc>().add(const SyncFavouritePlaces());
     context.read<LocationHistoryBloc>().add(const SyncLocationHistory());
-    context.read<KyaBloc>().add(const FetchQuizzes());
```
Comment (Baalmart) [inline mobile/lib/screens/dashboard/dashboard_view.dart:576]: Hi @BenjaminSsempala , please tell me more about this removed code. Just share as if you are onboarding me. Thanks!

**Review by Baalmart — COMMENTED** [review]
Hi @BenjaminSsempala , I have just left a small comment for your review. Thanks!

File: mobile/lib/screens/dashboard/dashboard_view.dart (hunk 2 above)
Comment (BenjaminSsempala) [reply mobile/lib/screens/dashboard/dashboard_view.dart:576]: Hello @Baalmart 
I'd added this in the translation PR such that the lessons and Quiz are translated immediately when language is changed, in that when the language is changed, the quiz endpoint is called again with the new parameter for language, something like `{AirQoUrls.kya}/quizzes/users/$userId?language=fr`
This introduced a bug of sorts, in that the quiz/kya was being fetched continuously before the progress on the app was synced, so the progress and progress bar were behaving unexpectedly. 

The solution I came up with was to clear the KYA and Quiz and then recall them, which would only be done at startup or restart of the app, and the change for that is also in this PR
` context.read<KyaBloc>().add(const ClearKya());
    context.read<KyaBloc>().add(const ClearQuizzes());`

Comment (Baalmart) [reply mobile/lib/screens/dashboard/dashboard_view.dart:576]: thanks @BenjaminSsempala

**Review by Baalmart — APPROVED** [review]

**Pull request merged**